crates/js/lib.rs

```diff
@@ -216,10 +217,37 @@ impl Runtime {
         .to_object(scope)
         .ok_or(Error::Terminated)?;
       let state_key = v8::String::new(scope, "state").unwrap().into();
-      let state_obj = state.get(scope, state_key).unwrap();
-      self.contract_state = v8::Global::new(scope, state_obj);
+      let state_obj_maybe = state.get(scope, state_key);
+      let result_key = v8::String::new(scope, "result").unwrap().into();
+      let result_obj_maybe = state.get(scope, result_key);
+
+      let is_result_some = result_obj_maybe.is_some();
+      let is_state_some = state_obj_maybe.is_some();
+      let mut curr_state: Option<Local<deno_core::v8::Value>> = None;
+      if is_state_some || is_result_some {
+        let state_str = v8::String::new(scope, "state").unwrap().into();
+        if is_result_some {
```

Comment:
L#229 TO L#246 
![image](https://user-images.githubusercontent.com/60560109/148317304-4bfcfbe5-5548-423c-b8d5-d99f9201669d.png)